Update I18n/Time.php, fix timeAgoInWords(timezone)

icaroscherma opened the pull request:
Fixed issue #6388 with the timezone not being applied in Cake\I18n\Time\timeAgoInWords().

icaroscherma committed:
Update I18n/Time.php, fix timeAgoInWords(timezone)

Fixed issue with the timezone not being applied in Cake\I18n\Time\timeAgoInWords().

markstory commented:
Would we be able to get a test case change to ensure this doesn't get broken again?

markstory commented:
I think we should probably restore the original timezone at the end of this method as well. Mutating the time objects when formatting is not going to be expected behavior.

icaroscherma commented:
I will take a look, I'm not very familiar with the test cases yet (unfortunately, I know, but I'm trying to change this). As soon as I have a solid feedback I will post it here.

markstory commented:
The example you posted in the original issue is a really great start to a test case. You can use the [existing tests](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/I18n/TimeTest.php#L335-L339) for formatting as a starting point as well.

icaroscherma committed:
Update TimeTest.php, add testTimeAgoInWordsTimezone

Added a test case for the timezone option in function timeAgoInWords.

icaroscherma committed:
Merge pull request #1 from icaroscherma/patch-2

Update TimeTest.php, add testTimeAgoInWordsTimezone

icaroscherma commented:
I added a Test Case, not sure if this is what you wanted or if I need more tests to be suitable for a merge... BTW: the "random" date that I used is my birthday. =)

markstory commented:
Test case looks good. There should also probably be an assertion to make sure the original time object didn't have its timezone permanently changed.

icaroscherma committed:
Cloning $this instead of changing timezone directly

Using a clone from ``$this`` in ``timeAgoInWords()`` the same way used in ``i18nFormat()``

icaroscherma commented:
Ahhh, after some tests locally here I figured out what you meant. I cloned ``$this`` if ``$timezone`` is set like in ``i18nFormat()``. File already changed. =)

antograssiot commented:
@icaroscherma you have some [CS issue on Travis](https://travis-ci.org/cakephp/cakephp/jobs/59489508)

icaroscherma committed:
Fixed space indents (Travis-CI warned me)

Fixed some space indents

markstory commented:
Thanks @icaroscherma :+1:
